### ISSUE TITLE
Add Mobile Apps to onboarding

### ISSUE THREAD

**Issue by SeanPM5** [home-assistant/home-assistant-polymer#4312]
Now that both platforms have an official mobile app, would be handy to get users situated with them during the onboarding process imo. This would be an optional / skippable step. 

balloob had a genius idea a few days ago in the ios Discord room about scanning a QR code to get the mobile apps setup. I think that would be awesome. But until that gets implemented, I think it could at least link to the mobile apps. 

Here is a quick and ugly mockup.

![onboarding-mobile](https://i.imgur.com/wcOzfyG.jpg)

**Comment by frenck** [comment]
oh, I like that one!

**Comment by iantrich** [comment]
http://qrappdownload.appspot.com/

**Comment by springstan** [comment]
http://onelink.to/

This service seems to provide the same functionality. However, so far I could only test the generated qr code with an android device.

**Comment by robbiet480** [comment]
FYI we do have [Smart App Banners](https://developer.apple.com/library/archive/documentation/AppleApplications/Reference/SafariWebContent/PromotingAppswithAppBanners/PromotingAppswithAppBanners.html) enabled for iOS (#3100).

I just did the first half of the work required to enable similar functionality on Android at https://github.com/home-assistant/home-assistant/pull/31808

**Comment by robbiet480** [comment]
Oh yeah, on the Android side for whatever reason the prompt only works if contents are served over HTTPS. Lame

**Comment by balloob** [comment]
btw I don't think that we should add mobile apps to our onboarding. The user does not know what Home Assistant is about so will not be interested in installing apps. Instead, we should promote the mobile apps when they add a person or maybe on the config dashboard if there are no `mobile_app` config entries (dismissable ofc)

**Comment by iantrich** [comment]
> The user does not know what Home Assistant is about so will not be interested in installing apps

I don't think this is likely the case. It does and will, for the foreseeable future, take research/effort to setup a new installation of Home Assistant. In doing so they will inevitably have a reason for doing so and know enough. Further, the smart home of current is app-driven, if anything it is expected to be presented with an app to use IMO, especially for a casual user.

**Comment by SeanPM5** [comment]
> Instead, we should promote the mobile apps when they add a person or maybe on the config dashboard if there are no `mobile_app` config entries (dismissable ofc)

I get the logic behind this, but with onboarding 100% of users will see it. Even if they don't want to commit to installing an app at that time, they're at least made aware of the _existence_ of apps. Some users might never add a person, and even the config dash link might be overlooked. These are good ideas though, I just think they should be done _in addition to_ rather than in place of this idea.

IMO the sooner that users get the apps, the better. They'd get a device tracker right away which opens up all the presence detection automations. And it also leads to them signing up for Cloud so they could get remote access, etc etc. One thing leads to another in a really organic way. 

Users doing this just on a desktop web browser wouldn't really have that same sense of direction. They'd be wondering where to start, perhaps getting frustrated and giving up sooner.
